runtime/autoload/health.vim

```diff
@@ -1,433 +1,181 @@
-                \ . 'This could mean that a) the "pyenv" executable is not in '
-                \ . '$PATH, b) your pyenv installation is broken.  '
-                \ . 'You should set "g:%s" to avoid surprises.',
-                \ python_bin, host_prog_var))
-        endif
-      endif
-    endif
-  endif
-
-  if !empty(python_bin)
-    if !empty(pyenv) && !exists('g:'.host_prog_var) && !empty(pyenv_root)
-          \ && resolve(python_bin) !~# '^'.pyenv_root.'/'
-      call add(notes, printf('Suggestion: Create a virtualenv specifically '
-            \ . 'for Neovim using pyenv and use "g:%s".  This will avoid '
-            \ . 'the need to install Neovim''s Python client in each '
-            \ . 'version/virtualenv.', host_prog_var))
-    endif
-
-    if !empty(venv) && exists('g:'.host_prog_var)
-      if !empty(pyenv_root)
-        let venv_root = pyenv_root
-      else
-        let venv_root = fnamemodify(venv, ':h')
-      endif
-
-      if resolve(python_bin) !~# '^'.venv_root.'/'
-        call add(notes, printf('Suggestion: Create a virtualenv specifically '
-              \ . 'for Neovim and use "g:%s".  This will avoid '
-              \ . 'the need to install Neovim''s Python client in each '
-              \ . 'virtualenv.', host_prog_var))
-      endif
-    endif
-  endif
-
-  if empty(python_bin) && !empty(python_bin_name)
-    " An error message should have already printed.
-    call add(notes, printf('Error: "%s" was not found.', python_bin_name))
-  elseif !empty(python_bin) && !s:check_bin(python_bin, notes)
-    let python_bin = ''
-  endif
-
-  " Check if $VIRTUAL_ENV is active
-  let virtualenv_inactive = 0
-
-  if exists('$VIRTUAL_ENV')
-    if !empty(pyenv)
-      let pyenv_prefix = resolve(s:trim(system(printf('"%s" prefix', pyenv))))
-      if $VIRTUAL_ENV != pyenv_prefix
-        let virtualenv_inactive = 1
-      endif
-    elseif !empty(python_bin_name) && exepath(python_bin_name) !~# '^'.$VIRTUAL_ENV.'/'
-      let virtualenv_inactive = 1
-    endif
-  endif
-
-  if virtualenv_inactive
-    call add(notes, 'Warning: $VIRTUAL_ENV exists but appears to be '
-          \ . 'inactive.  This could lead to unexpected results.  If you are '
-          \ . 'using Zsh, see: http://vi.stackexchange.com/a/7654/5229')
+""
+" s:add_single_checker is a function to handle adding a checker of name
+" {checker_name} to the list of health_checkers. It also enables it.
+function! s:add_single_checker(checker_name) abort
+  if has_key(g:health_checkers, a:checker_name)
+    " TODO: What to do if it's already there?
+    return
+  else
+      let g:health_checkers[a:checker_name] = v:true
   endif
+endfunction
 
-  " Diagnostic output
-  echo 'Checking: Python' a:version
-  echo '  Executable:' (empty(python_bin) ? 'Not found' : python_bin)
-  if len(python_multiple)
-    for path_bin in python_multiple
-      echo '     (other):' path_bin
+""
+" s:add_checker is a function to register at least one healthcheckers.
```

Comment:
The phrase "s:add_checker is a function to" is not necessary. In general, comments don't need to introduce the thing they're describing; it's obvious that this is describe the `s:add_checker` thing, so it doesn't need to be said. Instead, this is the only comment that is needed:

    " Registers at least one healthchecker.

This advice applies to all comments, there are other cases below and above.